src/main/java/org/apidesign/polyfill/websocket/WebSocketPolyfill.java

```diff
@@ -18,110 +23,98 @@
 import io.helidon.websocket.WsUpgradeException;
 
 public final class WebSocketPolyfill {
+
+    private static final String NEW_WEB_SOCKET_SERVER_DATA = "new-web-socket-server-data";
+    private static final String SET_INTERVAL = "set-interval";
+    private static final String CLEAR_INTERVAL = "clear-interval";
+    private static final String SET_TIMEOUT = "set-timeout";
+    private static final String CLEAR_TIMEOUT = "clear-timeout";
+
+    private static final String WEBSOCKET_POLYFILL_JS = "websocket-polyfill.js";
+
     private WebSocketPolyfill() {
     }
 
-    public static void prepare(Context ctx) {
-        var code = """
-        (function (jvm) {
-            var timer = 0;
-
-            globalThis.clearTimeout = function() {
-                debugger;
-            }
-            globalThis.clearInterval = function() {
-                debugger;
-            }
-            globalThis.setInterval = function() {
-                debugger;
-            }
-            globalThis.setTimeout = function(fn, delay, arg1, arg2, arg3) {
-                debugger;
-                return ++timer;
-            }
+    public static CompletableFuture<Context> prepare(Supplier<Context> contextSupplier, ExecutorService executor) {
+        CompletableFuture<Context> futureContext = new CompletableFuture<>();
+        Timers timers = new Timers(executor);
+        Source polyfill = Source
+                .newBuilder("js", WebSocketPolyfill.class.getResource(WEBSOCKET_POLYFILL_JS))
+                .buildLiteral();
 
-            globalThis.crypto = {
-                subtle : 0,
-                randomUUID : function() {
-                   debugger;
-                   throw 'randomUUID';
-                },
-                getRandomValues : function(arr) {
-                   for (let i = 0; i < arr.length; i++) {
-                     arr[i] = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
-                   }
-                   return arr;
+        executor.execute(() -> {
+            Context ctx = contextSupplier.get();
```

Comment:
OK, so all the _events_ are going to be processed in `executor`. That's OK I think.